staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -136,7 +136,12 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			Categories:   crd.Status.AcceptedNames.Categories,
 		})
 
-		if crd.Spec.Subresources != nil && crd.Spec.Subresources.Status != nil {
+		// TODO(roycaihw): Schema, Subresources and AdditionalPrinterColumns: these three
+		// fields should not be set for first item in Versions list. This is
+		// for backwards compatibility and we will fix this API in apiextensions.v1.
+		i := apiextensions.GetCRDVersionIndex(crd, version.Version)
```

Comment:
Instead of `GetCRDVersionIndex` method I suggest a `GetCRDSubresourceForVersion` function which returns the subresource based on the version.